zkevm-circuits/src/mpt_circuit.rs

```diff
@@ -285,502 +385,233 @@ impl<F: FieldExt> MPTConfig<F> {
+                let account_leaf_key_c;
+                ifx!{f!(position_cols.q_enable) => {
+                    ifx!{a!(account_leaf.is_key_s) => {
+                        account_leaf_key_s = AccountLeafKeyConfig::configure(meta, &mut cb, ctx.clone(), true);
+                    }}
+                    ifx!{a!(account_leaf.is_key_c) => {
+                        account_leaf_key_c = AccountLeafKeyConfig::configure(meta, &mut cb, ctx.clone(), false);
+                    }}
+                }}
+                // ACCOUNT_NON_EXISTING
+                let account_non_existing;
+                ifx!{f!(position_cols.q_enable), a!(account_leaf.is_non_existing) => {
+                    account_non_existing = AccountNonExistingConfig::configure(meta, &mut cb, ctx.clone());
+                }}
+                // ACCOUNT_LEAF_NONCE_BALANCE_S
+                // ACCOUNT_LEAF_NONCE_BALANCE_C
+                let account_leaf_nonce_balance_s;
+                let account_leaf_nonce_balance_c;
+                ifx!{f!(position_cols.q_not_first) => {
+                    ifx!{a!(account_leaf.is_nonce_balance_s) => {
+                        account_leaf_nonce_balance_s = AccountLeafNonceBalanceConfig::configure(meta, &mut cb, ctx.clone(), true);
+                    }}
+                    ifx!{a!(account_leaf.is_nonce_balance_c) => {
+                        account_leaf_nonce_balance_c = AccountLeafNonceBalanceConfig::configure(meta, &mut cb, ctx.clone(), false);
+                    }}
+                }}
+                // ACCOUNT_LEAF_STORAGE_CODEHASH_S
+                // ACCOUNT_LEAF_STORAGE_CODEHASH_C
+                let account_leaf_storage_codehash_s = AccountLeafStorageCodehashConfig::configure(meta, &mut cb, ctx.clone(), true);
+                let account_leaf_storage_codehash_c = AccountLeafStorageCodehashConfig::configure(meta, &mut cb, ctx.clone(), false);
+                // ACCOUNT_DRIFTED_LEAF
+                let account_leaf_key_in_added_branch;
+                ifx!{f!(position_cols.q_not_first), a!(position_cols.not_first_level), a!(account_leaf.is_in_added_branch) => {
+                    account_leaf_key_in_added_branch = AccountLeafKeyInAddedBranchConfig::configure(meta, &mut cb, ctx.clone());
+                }}
+
+                /* Range checks */
+                // These range checks ensure that the value in the RLP columns are all byte value.
+                // These lookups also enforce the value to be zero if the passed in length is < 0.
+                // TODO(Brecht): would be safer/cleaner if this can be enabled everywhere even for branch child rlp1
+                // TODO(Brecht): do 2 bytes/lookup when circuit height >= 2**21
+                ifx!{f!(position_cols.q_enable) => {
+                    // Sanity checks (can be removed, here for safety)
+                    require!(cb.length_s.sum_conditions() => bool);
+                    require!(cb.length_c.sum_conditions() => bool);
```

Comment:
As above - I think these two constraints (sum is boolean) are needed and also each of the conditions needs to be boolean.